[ondretann] iP

ondretann opened the pull request:
### Thames
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Thames frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use
All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
And it is FREE!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

ondretann committed:
Added greeting and exit messages

ondretann committed:
Added echo

ondretann committed:
Added add and list

ondretann committed:
Added task file, added mark as done/not done

ondretann committed:
Added todo, event, deadline

ondretann committed:
Added exception handling

ondretann committed:
no message

ondretann committed:
Add automated testing for chatbot

ondretann committed:
no message

ondretann committed:
Add file save and file initialization

ondretann committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Add file save and file initialization

ondretann committed:
Add localtime format for all tasks and fix several file initialization and saving bugs within duke

ondretann committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add localtime format for all tasks and fix several file initialization and saving bugs within duke

ondretann committed:
Add Storage, TaskList and Ui classes for more OOP. Edit Duke to have make use of these additional classes

ondretann committed:
Add AddCommand, DeleteCommand, EditCommand and ExitCommand that inherits from abstract class Command. Add ThamesException class to handle cases where the chatbot cannot decipher the message input by user.

ondretann committed:
Improve exception handling with more edge cases and rename Duke to Thames.

ondretann committed:
Merge branch 'branch-A-MoreOOP'

* branch-A-MoreOOP:
  Improve exception handling with more edge cases and rename Duke to Thames.
  Add AddCommand, DeleteCommand, EditCommand and ExitCommand that inherits from abstract class Command. Add ThamesException class to handle cases where the chatbot cannot decipher the message input by user.
  Add Storage, TaskList and Ui classes for more OOP. Edit Duke to have make use of these additional classes

ondretann committed:
Add all files into packages for organisation. Shift all commands into Thames.command package and all tasks into Thames.task package.

ondretann committed:
Merge branch 'branch-A-Packages'

* branch-A-Packages:
  Add all files into packages for organisation. Shift all commands into Thames.command package and all tasks into Thames.task package.

ondretann committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  Bump gradle and lib version
  Add Gradle support
  docs/README.md: Tweak document template

ondretann committed:
Add Parser and TaskList JUnit testcases. Add equals method for all tasks for easier implementation of test cases.

ondretann committed:
Merge branch 'branch-A-JUnit'

* branch-A-JUnit:
  Add Parser and TaskList JUnit testcases. Add equals method for all tasks for easier implementation of test cases.

ondretann committed:
Modify build.gradle to allow packaging of application into a JAR file.

ondretann committed:
Merge branch 'branch-A-Jar'

* branch-A-Jar:
  Modify build.gradle to allow packaging of application into a JAR file.

ondretann committed:
Modify test method names and import statement orderings to follow the given java coding standard.

ondretann committed:
Add JavaDocs for non trivial methods and classes.

ondretann committed:
Add find to allow users to search tasks in task list.

ondretann committed:
Merge branch 'branch-level-9'

* branch-level-9:
  Add find to allow users to search tasks in task list.

ondretann committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add JavaDocs for non trivial methods and classes.

ondretann committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Modify test method names and import statement orderings to follow the given java coding standard.

Bandov commented on src/main/java/Thames/Parser.java:
great use of toLowerCase to ensure that the input passed in is in the correct format and its something that I did not think about!

Bandov reviewed (commented):
The codebase is exceptionally well-organized, demonstrating a strong adherence to Object-Oriented Programming principles and naming conventions. This review aims to provide constructive feedback to further refine and enhance the codebase, despite its already high quality.

Strengths
Adherence to OOP Principles: The code excellently utilizes encapsulation, inheritance, and polymorphism, making the system flexible and scalable. The use of interfaces and abstract classes promotes loose coupling and high cohesion across modules.

Naming Conventions and Readability: The naming conventions for classes, methods, and variables are meticulously followed, enhancing code readability and maintainability. It's clear that thoughtful consideration was put into naming, making it easy to understand the purpose and functionality of different parts of the codebase.

Code Structure and Organization: The project's structure is logical and well-organized, facilitating easy navigation through the codebase. The separation of concerns is evident, with distinct layers for data access, business logic, and user interface components.

Documentation and Comments: The documentation is comprehensive and concise, providing valuable insights into the design decisions and functionality of various components. Inline comments are judiciously used, adding clarity to complex logic without cluttering the code.

Testing and Reliability: The comprehensive suite of unit and integration tests demonstrates a commitment to reliability and quality assurance. Tests are well-written and cover a broad spectrum of scenarios, ensuring that the code behaves as expected under various conditions.

jayllo-c commented on build.gradle:
```diff
@@ -0,0 +1,42 @@
+plugins {
+    id 'java'
+    id 'application'
+    id 'com.github.johnrengelman.shadow' version '7.1.2'
+}
```
would adding the checkstyle plugin be useful to you in ensuring the code is in line with the coding standard?

jayllo-c commented on src/main/java/Thames/Parser.java:
```diff
@@ -0,0 +1,85 @@
+package Thames;
+
+import Thames.command.;
```
Do you think we need to explicitly import all 
classes, or is importing the package sufficient?

jayllo-c commented on src/main/java/Thames/Parser.java:
```diff
@@ -0,0 +1,85 @@
+        switch(input.toLowerCase()) {
+        case "todo":
+            if (split.length != 2) {
+                throw new ThamesException("Please provide your Todo task in the following format: \n" +
```
When wrapping around, ensure that symbols such as '+', ':', etc. are at the start of the following line. This issue is quite consistent throughout your code. Maybe consider using the Gradle checkstyle plugin.

jayllo-c commented on src/main/java/Thames/Parser.java:
```diff
@@ -0,0 +1,85 @@
+        String[] split = fullCommand.trim().split(" ", 2);
+        String input = split[0];
+
+        switch(input.toLowerCase()) {
```
Proper indentation of switch here, good job!

jayllo-c commented on src/main/java/Thames/Storage.java:
```diff
@@ -0,0 +1,67 @@
+package Thames;
```
The coding standard looks fine here!

jayllo-c commented on src/main/java/Thames/TaskList.java:
```diff
@@ -0,0 +1,101 @@
+        this.taskList = new ArrayList<>();
+        while (sc.hasNextLine()) {
+            String task = sc.nextLine();
+            String[] split = task.split(",");
```
Should you use the singular form or plural for the names representing a collection of objects?

jayllo-c commented on src/main/java/Thames/Thames.java:
```diff
@@ -0,0 +1,59 @@
+
+import java.io.FileNotFoundException;
+
+import Thames.command.Command;
```
Do you think we need to explicitly import all
classes, or is importing the package sufficient?

jayllo-c commented on src/main/java/Thames/Ui.java:
```diff
@@ -0,0 +1,119 @@
+     * Outputs error message when saved task list cannot be loaded or when no task list has been created before.
+     */
+    public void showLoadingError() {
+        System.out.println("It seems like you do not have a saved task list. I will be creating an empty one for you.");
```
For long lines, what is the desired length limit before we start doing line wrapping?

jayllo-c reviewed (commented):
Overall, minimal coding standard mistakes were found in the code. To catch all the small details I may have missed, install the checkstyle plugin and import the config files from the AB3 project as stated in the checkstyle guide here: https://se-education.org/guides/tutorials/checkstyle.html

Hopefully the comments helped!

ondretann committed:
Add GUI for Thames chatbot

ondretann committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add GUI for Thames chatbot

ondretann committed:
Add assert to ensure the code functions as expected.

ondretann committed:
Merge pull request #2 from ondretann/branch-A-Assertions

Add assert to ensure the code functions as expected.

ondretann committed:
Refactor package name into lowercase to follow Java naming convention and erase unused code

ondretann committed:
Merge branch 'branch-A-CodeQuality'

* branch-A-CodeQuality:
  Refactor package name into lowercase to follow Java naming convention and erase unused code

ondretann committed:
The chat bot currently does not support displaying the list of events happening on a particular day.

Displaying the schedule for a specified date will be useful for most users.
The added convenience allows them to plan their activities for the day and
hence, more room for productivity and organisation.

Users can execute the function by entering "schedule <date>" where the date is in
yyyy-mm-dd format, e.g. schedule 2024-03-01.

ondretann committed:
Merge branch 'branch-BCD-Extension'

* branch-BCD-Extension:
  The chat bot currently does not support displaying the list of events happening on a particular day.

ondretann committed:
Update README.md to display user guide

ondretann committed:
Update user guide to include date formatting

ondretann committed:
Update README.md to display user guide

ondretann committed:
Add Ui showcase

ondretann committed:
Fix bugs regarding the exiting of application via
"bye" command

ondretann committed:
Fix bugs for installation on different folder
